libs/wingc/src/fold.rs

```diff
@@ -0,0 +1,438 @@
+use crate::ast::{
+	ArgList, CatchBlock, Class, ClassField, Constructor, ElifBlock, Expr, ExprKind, FunctionBody, FunctionDefinition,
+	FunctionParameter, FunctionSignature, FunctionTypeAnnotation, Interface, InterpolatedString, InterpolatedStringPart,
+	Literal, Reference, Scope, Stmt, StmtKind, Symbol, TypeAnnotation, UserDefinedType,
+};
+
+/// Similar to the `visit` module in `wingc` except each method takes ownership of an
+/// AST node instance of a reference to it, and returns a new AST node instance.
+/// This trait can be useful for AST transformations.
```

Comment:
```suggestion
/// Similar to the `visit` module in `wingc` except each method takes ownership of an
/// AST node instead of a reference to it, and returns a new AST node instance.
/// This trait can be useful for AST transformations.
```